src/server/req_stat.c

```diff
@@ -691,8 +692,43 @@ status_node(struct pbsnode *pnode, struct batch_request *preq, pbs_list_head *ps
 	return (rc);
 }
 
+/**
+ * @brief
+ * 	encode_isrunhook - encode ATTR_has_runjob_hook attribute
+ *
+ * @param[in] patr - ptr to attribute to decode
+ * @param[in] name - attribute name
+ * @param[in] rescn - resource name or null
+ * @param[out] val - string holding values for attribute structure
+ *
+ * @retval      int
+ * @retval      0       if ok
+ * @retval      >0      error number1 if error,
+ * @retval      *patr   members set
+ *
+ */
+
+/*ARGSUSED*/
 
+static void
+update_isrunhook(attribute *pattr)
+{
+	hook *phook = NULL;
 
+	pattr->at_val.at_long = 0;
+
+	/* Check if there are any valid runjob hooks */
+	for (phook = (hook *) GET_NEXT(svr_runjob_hooks);
+	     phook != NULL;
+	     phook = (hook *) GET_NEXT(phook->hi_runjob_hooks)) {
+		if (phook->enabled) {
+			pattr->at_val.at_long = 1;
+			break;
+		}
+	}
+
+	pattr->at_flags |= ATR_VFLAG_SET | ATR_VFLAG_MODCACHE;
```

Comment:
I'd rather you check if you need to change it before you change it.  This way you won't have to invalidate the cache all the time.